ext/native/thin3d/thin3d_vulkan.cpp

```diff
@@ -1277,18 +1350,124 @@ uint32_t VKContext::GetDataFormatSupport(DataFormat fmt) const {
 	}
 }
 
+// Simple independent framebuffer image. Gets its own allocation, we don't have that many framebuffers so it's fine
+// to let them have individual non-pooled allocations.
+struct VKImage {
+	VkImage image;
+	VkImageView view;
+	VkDeviceMemory memory;
+	VkImageLayout layout;
+};
+
+void CreateImage(VulkanContext *vulkan, VKImage &img, int width, int height, VkFormat format, VkImageLayout initialLayout, bool color) {
+	VkImageCreateInfo ici{ VK_STRUCTURE_TYPE_IMAGE_CREATE_INFO };
+	ici.arrayLayers = 1;
+	ici.mipLevels = 1;
+	ici.extent.width = width;
+	ici.extent.height = height;
+	ici.extent.depth = 1;
+	ici.initialLayout = VK_IMAGE_LAYOUT_UNDEFINED;
+	ici.imageType = VK_IMAGE_TYPE_2D;
+	ici.samples = VK_SAMPLE_COUNT_1_BIT;
+	ici.tiling = VK_IMAGE_TILING_OPTIMAL;
+	ici.format = format;
+	ici.usage = VK_IMAGE_USAGE_SAMPLED_BIT | VK_IMAGE_USAGE_TRANSFER_DST_BIT | VK_IMAGE_USAGE_TRANSFER_SRC_BIT;
+	if (color) {
+		ici.usage |= VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT;
+	} else {
+		ici.usage |= VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT;
+	}
+	vkCreateImage(vulkan->GetDevice(), &ici, nullptr, &img.image);
+
+	VkMemoryRequirements memreq;
+	vkGetImageMemoryRequirements(vulkan->GetDevice(), img.image, &memreq);
+	
+	VkMemoryAllocateInfo alloc{ VK_STRUCTURE_TYPE_MEMORY_ALLOCATE_INFO };
+	alloc.allocationSize = memreq.size;
+	vulkan->MemoryTypeFromProperties(memreq.memoryTypeBits, VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT, &alloc.memoryTypeIndex);
+	VkResult res = vkAllocateMemory(vulkan->GetDevice(), &alloc, nullptr, &img.memory);
+	assert(res);
+	res = vkBindImageMemory(vulkan->GetDevice(), img.image, img.memory, 0);
+	assert(res);
```

Comment:
We should at some point move away from using `assert` here in case of out of memory, at least.  For later.

But should these be `res == VK_SUCCESS`?

-[Unknown]